### PR TITLE
Update guide to use JSON-B instead of Jackson

### PR DESCRIPTION
**Pull request by KyleAure**
Remaining changes requested at the end of the previous pull request here: https://github.com/OpenLiberty/guide-rest-client-java/pull/20

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -7,32 +7,33 @@
 //     IBM Corporation
 :projectid: rest-client-java
 :page-layout: guide
-:page-duration: 20 minutes
-:page-releasedate: 2017-10-04
+:page-duration: 25 minutes
+:page-releasedate: 2018-06-01
```
Comment (yeekangc) [inline README.adoc]: Need to be updated to when the guide will be refreshed.

File: README.adoc [hunk 2]
```diff
@@ -7,32 +7,33 @@
...
 :page-description: Learn how to consume a RESTful web service
```
Comment (yeekangc) [inline README.adoc]: Add "in Java using JSON-B and JSON-P" to the end.

File: start/src/main/java/io/openliberty/guides/consumingrest/model/Album.java [hunk 3]
```diff
@@ -12,36 +12,32 @@
...
+    
+    @JsonbCreator
+    public Album(
+    		@JsonbProperty("title") String title, 
+    		@JsonbProperty("artist") String artistName, 
+    		@JsonbProperty("ntracks") int totalTracks) {
+    	//or custom constructor can be used
+    	this.title = title;
+    	this.artistName = artistName;
+    	this.totalTracks = totalTracks;
```
Comment (yeekangc) [inline start/src/main/java/io/openliberty/guides/consumingrest/model/Album.java]: Alignment or extra whitespaces need to be fixed?

File: start/src/main/java/io/openliberty/guides/consumingrest/model/Artist.java [hunk 4]
```diff
@@ -12,31 +12,30 @@
...
-    
-    public void setAlbums(Album[] albums) {
-        this.albums = albums;
+    @JsonbCreator
+    public Artist(
+    		@JsonbProperty("name") String name, 
+    		@JsonbProperty("albums") Album albums[]) {
+    	//or custom constructor can be used
+    	this.name = name;
+    	this.albums = albums;
```
Comment (yeekangc) [inline start/src/main/java/io/openliberty/guides/consumingrest/model/Artist.java]: Alignment or whitespaces to be fixed?

File: start/src/main/java/io/openliberty/guides/consumingrest/service/ArtistResource.java [hunk 5]
```diff
@@ -22,6 +22,7 @@
 import javax.ws.rs.core.UriInfo;
 
 import io.openliberty.guides.consumingrest.model.Artist;
+//import io.openliberty.guides.consumingrest.Consumer;
```
Comment (yeekangc) [inline start/src/main/java/io/openliberty/guides/consumingrest/service/ArtistResource.java]: To be removed if not in use?

**Comment by yeekangc** [comment]
@KyleAure, looks like the latest changes are here including the fixes for #26.  Kyle, can you please doubly confirm?

@JunyiSun, can you confirm that we have this PR tested and both the guide and code check out?

Then, we can approve, merge and ask @kinueng to refresh the site for us.

@evelinec, FYI.

**Comment by KyleAure** [comment]
@yeekangc Just double checked and ran the tests again.  Warnings have been taken care of and the correct imports have been updated for ArtistResource.  Everything in #26 has been corrected.  Thank you.

**Comment by yeekangc** [comment]
We are good to go.  Thank you, @KyleAure.

**Pull request merged**